### DISCUSSION TITLE
How to make EVTV ESP32 CanDue V2 + SavvyCan work on Mac

### DISCUSSION THREAD

**Discussion by Chunkulator** [collin80/SavvyCAN#677] [General]
Hi Collin,

I've had no luck getting the above combination of stuff to work using MacOSX Monterey 12.6.3. When I set up the GVRET device in SavvyCan the status column in the Connection Settings window shows "Not Connected". I did manage to open it as a LAWICEL device, but don't seem to be able to log any frames (though I can see activity on my CAN bus using a scope). So the core thing I'm trying to figure out is how to connect my CanDue ESP32 V2 as a GVRET device.

My first point of confusion is that the CanDue ESP32 V2 appears as four different USB devices:
- /dev/cu.usbserial-14200
- /dev/tty.usbserial-14200
- /dev/cu.usbserial-14201
- /dev/tty.usbserial-14201
and I'm not sure which of these I'm supposed to be communicating with for each of the following tasks:
- flashing ESP32RET using Arduino
- flashing ESP32RET using ESP32RET Firmware Updater (confusingly, updater.sh looks for /dev/tty.USB* instead of /dev/tty.usbserial*)
- logging frames in SavvyCan as a GVRet device.

Could you please start by clarifying which of the four devices I should be using and then suggest why I might be unsuccessful in having my CanDue V2 connect as a GVRET device?

Thanks,
Richard

**Comment by danjhunter** [comment]
I'm having the same exact issue on macOS Monterey. I was also unable to compile the firmware using Arduino IDE and ESP32RET.ino I was successful however running updater.command after I edited the file and changed the usbserial address to 'dev/cu.usbserial-1100' I intially had an error "a fatal error occurred: failed to connect to ESP32: Timed out waiting for packet header". To fix this I held down the BOOT/FLASH button on the board and that worked. Should I just switch to PC? Seems there is also issue #27 "unable to connect EVTV ESP32 Board to SavvyCAN after flashing with ESP32RET". @collin80 any thoughts?

**Reply by Chunkulator** [reply]
I was able to make it work by compiling SavvyCAN from source. I guess the latest binary on the website is out of date.

**Reply by danjhunter** [reply]
@Chunkulator i also got this working by using v213. Version208 on the Savvy CAN website doesn't work for MacOS. Thanks for your help!

**Reply by nicksaintjohn** [reply]
FYI, I was able to install the trial version of QT Creator (the IDE), and compile the latest build of SavvyCAN on MacOS (Sonoma). It's buggy, and crashes occasionally, but all extra connection options including QT Serial bus are now available for me. Have got traffic coming through using a PeakCAN device. The source needed a bit of tweaking IIRC, which may be why it crashes occasionally, but it's running and perfectly usable!
